src/dune_rules/melange_rules.ml

```diff
@@ -139,8 +154,14 @@ let add_rules_for_entries ~sctx ~dir ~expander ~dir_contents ~scope
         build_js ~dir ~loc:(Some loc) ~pkg_name ~module_system:mel.module_system
           ~dst_dir ~obj_dir ~sctx ~build_dir ~lib_deps_js_includes m)
   in
-  let ctx = Super_context.context sctx in
-  let stdlib_dir = ctx.Context.stdlib_dir in
+  let* stdlib_dir =
+    let* melc = Super_context.resolve_program sctx ~dir ~loc:None "melc" in
+    match melc with
+    | Error _ -> Memo.return None
```

Comment:
is `ctx.stdlib_dir` just plain wrong? Falling back to it would avoid making `stdlib_dir` an option type.

Reply:
Yes, I don't see how it can be used. It's causing those "inconsistent assumptions" errors for Javi.

Reply:
aha, understood. that makes total sense.